### DISCUSSION TITLE
redoing a plugin with different options

### DISCUSSION THREAD

**Discussion by tinker2much** [gitbls/sdm#236] [General]
I _configure_ general images with common programs/settings/options. I then _burn_ cards for many different pi's on the back of those bases, setting specific things in the burn like host name or dtoverlay lines in config.txt. Up till now, the plugins didn't overlap, i.e., a given plugin was only done in the configure or in the burn, but not in both. 

Now, I have a case where I generally want:
`B1: boot to console, requiring login`
but occasionally, I would want: 
`B4: boot to desktop, logging in automatically`

Is there any problem if I always invoke the raspiconfig plugin in the configure, and then also (but only sometimes) do it in the burn? Will I get the later value, the one from the burn, as I expect? 
`+--plugin raspiconfig:"boot_behaviour=B4" \`

**Comment by gitbls** [comment]
This will work, and you'll get the setting from the last change (the one from the burn).

FYI this particular setting is kept in /etc/sdm/assets/gfxbbh and is simply over-written when set again. During the First Boot process it is checked and if the file exists and has valid content, the last requested boot behavior is set.

There are a couple of plugins that have issues with being run more than once. For instance, re-running the `apps` plugin with the same `name=` parameter (or twice without the `name` parameter) will get an error. The `parted` plugin will likely fail if rerurn (but it's a burn plugin so it's difficult to run it multiple times).  Others may fail if the IMG settings on which they are dependent change (e.g., the script run by a `runscript` plugin does something that causes it to fail if rerun). 

Other than these few, most plugins are not sensitive to being rerun. Except for the ones you'll find that prove me wrong 🤪
